### PR TITLE
Add support for providing optional components

### PR DESCRIPTION
**Pull request by mszostok**
**Description**

Changes proposed in this pull request:

- Add support for providing optional components
- Refactor input builder package
- Add tmpl render for KymaConfigToGraphQL
- Add option to override ServiceManager credentials

**Related issue(s)**

- https://github.com/kyma-incubator/compass/issues/691

### CODE REVIEW

**Comment by mszostok** [comment]
more test coverage will be added in this issue: https://github.com/kyma-incubator/compass/issues/723

File: components/kyma-environment-broker/internal/broker/builder_test.go [hunk 1]
```diff
@@ -3,52 +3,82 @@ package broker
...
+		}).
+		Build()
 
 	// then
+	require.NoError(t, err)
 	assert.Equal(t, "azure", input.ClusterConfig.GardenerConfig.Provider)
 	assert.Equal(t, "azure-cluster", input.ClusterConfig.GardenerConfig.Name)
-}
+	assert.Equal(t, "azure-secret", input.ClusterConfig.GardenerConfig.TargetSecret)
+	assert.EqualValues(t, input.KymaConfig.Components, mappedComponentList)
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/internal/broker/builder_test.go]: it seems input.KymaConfig.Components is an actual value not expected

File: components/kyma-environment-broker/internal/runtime/provider.go [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+}
+
+func (r ComponentsListProvider) getManagedRuntimeComponents() ([]v1alpha1.KymaComponent, error) {
+	yamlFile, err := ioutil.ReadFile(r.managedRuntimeComponentsYAMLPath)
+	if err != nil {
+		return nil, errors.Wrap(err, "while reading YAML file with managed components mangedList")
+	}
+
+	var mangedList struct {
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/internal/runtime/provider.go]: mangedList -> managedList

Comment (mszostok) [reply components/kyma-environment-broker/internal/runtime/provider.go]: fixed

File: components/kyma-environment-broker/internal/broker/builder_test.go (hunk 1 above)
Comment (mszostok) [reply components/kyma-environment-broker/internal/broker/builder_test.go]: fixed

File: components/kyma-environment-broker/internal/broker/builder.go [hunk 3]
```diff
@@ -1,62 +1,248 @@
...
+	return nil
+}
+
+// applyTemporaryCustomization applies some additional information. This is only a temporary solution for MVP scenario.
+// Will be removed and refactored in near future.
+func (b *InputBuilder) applyTemporaryCustomization(in *gqlschema.ProvisionRuntimeInput) error {
+	switch b.planID {
+	case azurePlanID:
+		in.ClusterConfig.GardenerConfig.TargetSecret = b.provisioningConfig.AzureSecretName
+	case gcpPlanID:
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/internal/broker/builder.go]: make configurable. On prod GCP should be disabled now

**Comment by piotrmiskiewicz** [comment]
please, compare components names as case insensitive

**Pull request merged**